[Terra-Drop-Down-Button] add navigation between last and first items (#3771)
cerner/terra-core: packages/terra-dropdown-button/CHANGELOG.md

# Changelog

## Unreleased

> * Fixed

Reviewer: nit : 
```suggestion

* Fixed
```

> * Fixed
>    * Added logic to change the selection when we are on last option to first option and vice-versa

Reviewer 2: ```suggestion

* Changed
   * Updated selection logic to cycle from last option to first option and vice-versa
```

Author: Done.
